### ISSUE TITLE
Grid pattern on image

### ISSUE THREAD

**Issue by LuisPalluel** [leftthomas/SRGAN#112]
Hello,

I used this implementation (thanks for that) and got some grid pattern when zooming into the image (see image example).
The reconstruction of the global image is well done but this pattern is also present on the whole image.
By the way, it also happens when I just clone the repo and don't change anything.

I was wondering if somebody has already encountered this kind of pattern ?

Thanks in advance !

![image](https://user-images.githubusercontent.com/28446415/130594509-e7a81353-223a-4f8e-9c9c-cd3bcb0fb87d.png)

**Comment by leftthomas** [comment]
you could try to change the loss weight and train again.

**Comment by KevinChen880723** [comment]
@LuisPalluel, @Suanmd, 
Hello, I just meet the same problem and find it can be solved by modifying `UpsampleBlock()` in `model.py` as the followings:
```python
class UpsampleBLock(nn.Module):
    def __init__(self, in_channels, up_scale):
        super(UpsampleBLock, self).__init__()
        self.conv_identity_res = nn.Conv2d(in_channels, in_channels, kernel_size=3, padding=1)
        self.prelu = nn.PReLU()
        self.up_scale = up_scale

    def forward(self, x):
        x = self.conv_identity_res(x)
        x = F.interpolate(x, scale_factor=self.up_scale, mode='bicubic')
        x = self.conv_identity_res(x)
        x = self.prelu(x)
        return x
# class UpsampleBLock(nn.Module):
#     def __init__(self, in_channels, up_scale):
#         super(UpsampleBLock, self).__init__()
#         self.conv = nn.Conv2d(in_channels, in_channels * up_scale ** 2, kernel_size=3, padding=1)
#         self.pixel_shuffle = nn.PixelShuffle(up_scale)
#         self.prelu = nn.PReLU()

#     def forward(self, x):
#         x = self.conv(x)
#         x = self.pixel_shuffle(x)
#         x = self.prelu(x)
#         return x

```
You can check the explanations from [this article](https://distill.pub/2016/deconv-checkerboard/)!

Below figures are the results from the two different model (the left is from the modified one) :

<img width="1368" alt="圖片" src="https://user-images.githubusercontent.com/55487740/192085717-f04ea6e4-eb19-410c-8c9f-ae0b0f795a1b.png">

**Comment by Suanmd** [comment]
> @LuisPalluel, @Suanmd, Hello, I just meet the same problem and find it can be solved by modifying `UpsampleBlock()` in `model.py` as the followings:
> 
> ```python
> class UpsampleBLock(nn.Module):
>     def __init__(self, in_channels, up_scale):
>         super(UpsampleBLock, self).__init__()
>         self.conv_identity_res = nn.Conv2d(in_channels, in_channels, kernel_size=3, padding=1)
>         self.prelu = nn.PReLU()
>         self.up_scale = up_scale
> 
>     def forward(self, x):
>         x = F.interpolate(x, scale_factor=self.up_scale, mode='bicubic')
>         x = self.conv_identity_res(x)
>         x = self.prelu(x)
>         return x
> # class UpsampleBLock(nn.Module):
> #     def __init__(self, in_channels, up_scale):
> #         super(UpsampleBLock, self).__init__()
> #         self.conv = nn.Conv2d(in_channels, in_channels * up_scale ** 2, kernel_size=3, padding=1)
> #         self.pixel_shuffle = nn.PixelShuffle(up_scale)
> #         self.prelu = nn.PReLU()
> 
> #     def forward(self, x):
> #         x = self.conv(x)
> #         x = self.pixel_shuffle(x)
> #         x = self.prelu(x)
> #         return x
> ```
> 
> You can check the explanations from [this article](https://distill.pub/2016/deconv-checkerboard/)!
> 
> Below figures are the results from the two different model (the left is from the modified one) :
> 
> <img alt="圖片" width="1368" src="https://user-images.githubusercontent.com/55487740/192085717-f04ea6e4-eb19-410c-8c9f-ae0b0f795a1b.png">



> @LuisPalluel, @Suanmd, Hello, I just meet the same problem and find it can be solved by modifying `UpsampleBlock()` in `model.py` as the followings:
> 
> ```python
> class UpsampleBLock(nn.Module):
>     def __init__(self, in_channels, up_scale):
>         super(UpsampleBLock, self).__init__()
>         self.conv_identity_res = nn.Conv2d(in_channels, in_channels, kernel_size=3, padding=1)
>         self.prelu = nn.PReLU()
>         self.up_scale = up_scale
> 
>     def forward(self, x):
>         x = F.interpolate(x, scale_factor=self.up_scale, mode='bicubic')
>         x = self.conv_identity_res(x)
>         x = self.prelu(x)
>         return x
> # class UpsampleBLock(nn.Module):
> #     def __init__(self, in_channels, up_scale):
> #         super(UpsampleBLock, self).__init__()
> #         self.conv = nn.Conv2d(in_channels, in_channels * up_scale ** 2, kernel_size=3, padding=1)
> #         self.pixel_shuffle = nn.PixelShuffle(up_scale)
> #         self.prelu = nn.PReLU()
> 
> #     def forward(self, x):
> #         x = self.conv(x)
> #         x = self.pixel_shuffle(x)
> #         x = self.prelu(x)
> #         return x
> ```
> 
> You can check the explanations from [this article](https://distill.pub/2016/deconv-checkerboard/)!
> 
> Below figures are the results from the two different model (the left is from the modified one) :
> 
> <img alt="圖片" width="1368" src="https://user-images.githubusercontent.com/55487740/192085717-f04ea6e4-eb19-410c-8c9f-ae0b0f795a1b.png">

Hi, Thanks for your reply. @KevinChen880723 
Up-sampling modules like yours (by using F.interpolate()) seem to be rare. I also noticed a problem, please see [here](https://discuss.pytorch.org/t/artefacts-when-using-a-perceptual-loss-term/146064), do you think it is still a problem with the upsampling module?

**Comment by KevinChen880723** [comment]
Hi, @Suanmd.
I'm also new to super-resolution, thank u for sharing ur perspective!
But isn't the perceptual loss just used to make the high-level features of the real and fake images similar?

By the way, although changing the architecture can avoid the checkerboard artifacts, I find it makes the training became unstable. Sometimes the model just outputs whole black images, do u have any suggestions on this problem?
